File: IOT/app/src/main/java/com/example/velm/iot/MyApp.java
package com.example.velm.iot;

import android.app.Application;
import android.os.StrictMode;

import com.squareup.leakcanary.LeakCanary;

/**
 * Created by velmmuru on 10/4/2017.
 */

public class MyApp extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        setupLeakCanary();
    }

    private void setupLeakCanary() {
        if(LeakCanary.isInAnalyzerProcess(this)){
            return;
        } else {
            enabledStrictMode();
            LeakCanary.install(this);
        }
    }

    private void enabledStrictMode() {
        StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder().build());
    }
}

File: dragger2/src/main/java/app/com/dragger2/MyApp.kt
package app.com.dragger2

import android.app.Application
import android.content.Context

class MyApp : Application() {

    private var apiComponent: ApiComponent? = null

    operator fun get(context: Context): MyApp {
        return context.applicationContext as MyApp
    }

    override fun onCreate() {
        super.onCreate()
        apiComponent = DaggerApiComponent.builder()
                .appModule(AppModule(this))
                .apiModule(ApiModule())
                .build()
    }

    fun getApiComponents() : ApiComponent {
        return apiComponent!!
    }

}